### PR TITLE
Let bootstrap.py check out exact branches.

### PR DESCRIPTION
**Pull request by spxtr**

### CODE REVIEW

File: jenkins/bootstrap.py [hunk 1]
```diff
@@ -147,22 +147,22 @@ def _call(end, cmd, stdin=None, check=True, output=None):
     return lines
 
 
-def pull_has_shas(pull):
-    """Determine if a pull reference specifies shas (contains ':')"""
+def ref_has_shas(pull):
```
Comment (rmmh) [inline jenkins/bootstrap.py]: rename the arg from pull to ref

File: jenkins/bootstrap_test.py [hunk 2]
```diff
@@ -251,6 +251,12 @@ def testPullRef(self):
...
             (['+refs/pull/123/merge'], ['FETCH_HEAD']))
 
+    def testBranchRef(self):
+        self.assertEqual(bootstrap.branch_ref('branch:abcd'),
+            (['branch'], ['abcd']))
+        self.assertEqual(bootstrap.branch_ref('master'),
+            (['master'], ['FETCH_HEAD']))
+
 
 class CheckoutTest(unittest.TestCase):
```
Comment (rmmh) [inline jenkins/bootstrap_test.py]: Please add a test under IntegrationTest for your expected use case-- I'm not convinced that the current fetching code will do the right thing to make it able to be checked out.

File: jenkins/bootstrap.py (hunk 1 above)
Comment (spxtr) [inline jenkins/bootstrap.py]: Done.

File: jenkins/bootstrap_test.py (hunk 2 above)
Comment (spxtr) [inline jenkins/bootstrap_test.py]: Done.

**Pull request merged**